react/components/TrapFocus.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { useEffect, useRef } from 'react'
+
+interface Props {
+  children: React.ReactElement
+  open: boolean
+}
+
+export default function TrapFocus(props: Props) {
+  const { children, open } = props
+  const childRef = useRef<HTMLElement>(null)
+
+  useEffect(() => {
+    if (open && childRef.current) {
+      childRef.current.focus()
+    }
+  }, [open])
```

Comment:
`childRef` should also be a dependency, right?

Reply:
`useRef` always return the same object, so it never changes